File: code/session-1/python/interactive-example.py
low = 0
high = int(input())
state = "initial"
while state != "correct":
    mid = (high+low)//2
    _, state = print(mid, flush=True), input().strip()
    if state == "higher":
        low = mid+1
    if state == "lower":
        high = mid - 1

File: code/session-1/java/InteractiveExample.java
import java.io.*;

public class InteractiveExample {
  public static void main(String... args) throws IOException {
    var output = new BufferedWriter(new OutputStreamWriter(System.out));
    var input = new BufferedReader(new InputStreamReader(System.in));
    var low = 0;
    var high = Integer.parseInt(input.readLine());
    String state;
    do {
      var mid = (low + high) >> 1;
      output.write(mid + "\n");
      output.flush();
      state = input.readLine();
      if (state.equals("higher")) {
        low = mid + 1;
      } else if (state.equals("lower")) {
        high = mid - 1;
      }
    } while (!state.equals("correct"));
  }
}
